syncstorage-db-common/src/results.rs

```diff
@@ -86,6 +86,14 @@ pub struct ConnectionInfo {
 
 pub type GetCollectionId = i32;
 
+pub type GetOrCreateCollectionId = i32;
```

Comment:
let's get rid of the bits in here for a couple reasons:

- results/params only include the types needed for the trait method params/results, whereas both these types are more internal details. Especially for `GetOrCreateCollectionId`, if we really want a standard type to reference we could probably just reuse the existing `GetCollectionId`
- At some point it probably make sense to put shared diesel things like `IdResult` somewhere (maybe this justifies a `syncserver-diesel` crate? but I'm avoiding that for now as we have a lot of crates!). There is a `diesel.rs` module in syncstorage-db-common where this could live instead for the sake of sharing (`syncstorage-mysql` has an exact copy of this `IdResult`)